src/schema/firefox-schemas-import.js

```diff
@@ -354,7 +356,14 @@ export function rewriteExtend(schemas, schemaId) {
       const { $extend, id, ...rest } = type;
       if ($extend) {
         // Move the $extend into definitions.
-        extendDefinitions[$extend] = rest;
+        //
+        // NOTE: some schema files, like browser_action.json, may contains more than one extends
+        // and so here we use deepmerge to merge the multiple extended definitions instead
+        // of overwriting it with the last one processed.
+        extendDefinitions[$extend] = deepmerge(
+          extendDefinitions[$extend] ?? {},
+          rest
+        );
```

Comment:
I suppose this is for things like this: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/browser_action.json#50-68

Is there a chance that the `properties` keys are the same in the future? (which would lead to unexpected deep merges I'd suspect).

Reply:
> I suppose this is for things like this: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/browser_action.json#50-68

yes it is

> Is there a chance that the properties keys are the same in the future? (which would lead to unexpected deep merges I'd suspect).

it shouldn't happen, but honestly I think that it would be a good idea if we would explicitly check we are not overwriting a property with a different definition, and throw an explicit error if we are.
We will still need to review the imported schema as we are already doing, and we may likely notice that if it happens, but I would still feel more "relaxed" if we throw on importing an unexpected set of schema data, instead of silently doing something potentially unexpected.

I'm going to add an explicit "check and throw on invalid data" here.